### ISSUE TITLE
[Feature Proposal] Cheat Sheet

### ISSUE THREAD

**Issue by dericcain** [tailwindcss/tailwindcss#239]
I think it would be very helpful to have a cheat sheet for all of the utility classes and such. Since we have access to all of the utilities in the `.js` files, we could probably create a npm script that could generate the file to display all of the utilities.

Whatcha think?

**Comment by adamwathan** [comment]
This is definitely a great idea and something we've wanted to do from the beginning. Have you seen any other cheat sheets that you think are formatted in a really useful way that we could take inspiration from? Or are you just thinking of a giant table similar to this but with every class:

![image](https://user-images.githubusercontent.com/4323180/33382489-de09463e-d4ee-11e7-97e7-3330849d0139.png)

**Comment by tlgreg** [comment]
I also kinda like, what tachyons does with their overview pages, it's easy to look up a property.
[Table of Styles](http://tachyons.io/docs/table-of-styles/)
[Table of Properties](http://tachyons.io/docs/table-of-properties/)

**Comment by dericcain** [comment]
I really like what Tachyons is doing with theirs. When I thought of it, I was thinking more along the lines of this: http://fontawesome.io/cheatsheet/ With that said, I do not think it has to be just like that and I really like the layout that @adamwathan presented. 

it would be cool if it had a search mechanism. I mean, `cmd+f/ctrl+f` works, so that is not a necessity - rather, a luxury.

**Comment by psren** [comment]
Mhh, maybe it's possible to index all classes to algolia somehow and add a third section to autocomplete ore something like that.

**Comment by dericcain** [comment]
It doesn’t even have to be that in-depth, although, that would be killer. It could be as simple as using JS .filter() 

Deric Cain

**Comment by damian-nz** [comment]
Maybe even just simply grouping the classes by function

**Comment by chrisdmacrae** [comment]
@adamwathan could this be done outside of core?

**Comment by adamwathan** [comment]
Yep totally could I think; someone has actually made one here just not sure if it's autogenerated or put together by hand:

http://nerdcave.com/tailwind-cheat-sheet

**Comment by derekphilipau** [comment]
@ZachJMoore and I wanted to use Tailwind CSS in some projects and thought a cheat sheet would be a great way to learn:  https://github.com/derekphilipau/tailwindcss-cheatsheet-single-page  Zach also made some layout & component examples here:  https://github.com/ZachJMoore/tailwindcss-layout-examples  We hope you find them useful and let us know if you have ideas for improvements!

**Comment by donnysim** [comment]
It would be great if in addition to cheat sheet, tailwind could generate an html file where it displays how everything looks based on tailwind config colors, shadows etc., including ability to add to the generated html for plugins like buttons etc..
